tasks/create.js

```diff
@@ -55,45 +63,81 @@ module.exports = async function(args={}) {
     throw new Error(error.message);
   }
 
-  // Build migrations path
-  let dir = path.join(process.cwd(), value.directory);
+  // Proceed to create migrations based on types.
+  createStrategies[value.types.length](path.join(process.cwd(), value.directory), value);
+}
+
+/**
+ * 
+ * @param {*} config 
+ */
+function createSchemaOnlyMigrations(dir, config={}) {
 
   // Check if dir exists - if not make it
   let exists = fs.existsSync(dir);
 
   if(!exists) {
     fs.mkdirSync(dir);
-  };
+  }
+
+  let {name, renames} = migrationNameMappings[config.ordering](config.name.replace(/\s/gi, '-'), dir, 'schema', config.before);
+  let content = templates[config.mode](config.name, 'schema');
+
+  migrationFileMappings[config.ordering](name, dir, content, renames);
+}
 
-  let {name, renames} = migrationNameMappings[value.ordering](value.name.replace(/\s/gi, '-'), dir, value.before);
-  let content = template(value.name);
+/**
+ * 
+ * @param {*} config 
+ */
+function createSchemaAndDataMigrations(dir, config={}) {
+  // Check if dir exists - if not make it
+  let exists = fs.existsSync(dir);
 
-  migrationFileMappings[value.ordering](name, dir, content, renames);
+  if(!exists) {
+    fs.mkdirSync(dir);
+    // Create sub directories
+    config.types.forEach(t => fs.mkdirSync(path.join(dir, t)));
+  } else {
+    // Root directory exists, now make sure the two sub directories exist.
+    config.types.filter(t => !fs.existsSync(path.join(dir, t))).forEach(t => fs.mkdirSync(path.join(dir, t)));
+  }
+
+  config.types.forEach(type => {
+    let loc = path.join(dir, type);
+    let content = templates[config.mode](config.name, type);
+    let {name, renames} = migrationNameMappings[config.ordering](config.name.replace(/\s/gi, '-'), loc, type, config.before);
+    migrationFileMappings[config.ordering](name, loc, content, renames);
+  });
 }
 
 /**
  * [generateSequentialName description]
  * @param  {[type]} name [description]
  * @return {[type]}      <number>.<name>.migration.js
  */
-function generateSequentialName(name, dir, before) {
+function generateSequentialName(name, dir, type, before) {
   // Get array of all files in migration dir
   let files = fs.readdirSync(dir);
   let sequence = files.length && files.map((file='') => parseInt(file.split('.')[0] || 0)).sort((a, b) => a-b) || [0];
-  let beforeNext = -1;
+  let names = files.length && (files.map(file => file.split('.')[1])) || [];
   let renameList = [];
-  // If we are inserting an entry determine the correct index
+  let beforeNext = -1;
+  
   if (before) {
-    beforeNext = (files.findIndex(file => file.includes(before)));
+    beforeNext = names.indexOf(before);
```

Comment:
@jcroft-pdx I changed the before logic to use just the name of the migration to be consistent with the rest of the cli-api.